### PR TITLE
Refactor before fixing the TOML crashes

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Some small refactor prior to #4720 so it's easier to see what really changed in the final PR.

### CODE REVIEW

File: pylint/config/option_manager_mixin.py [hunk 1]
```diff
@@ -273,25 +274,7 @@ def read_config_file(self, config_file=None, verbose=None):
             parser = self.cfgfile_parser
```
Comment (DanielNoord) [inline pylint/config/option_manager_mixin.py:274]: Might as well remove the `disable` in this PR.

**Review by DanielNoord — APPROVED** [review]

File: pylint/config/option_manager_mixin.py [hunk 2]
```diff
@@ -311,6 +294,30 @@ def read_config_file(self, config_file=None, verbose=None):
             msg = "No config file found, using default configuration"
         print(msg, file=sys.stderr)
 
+    @staticmethod
+    def _parse_toml(
+        config_file: Union[Path, str], parser: configparser.ConfigParser
+    ) -> None:
+        """Upgrade the parser from a toml configuration file."""
```
Comment (DanielNoord) [inline pylint/config/option_manager_mixin.py:301]: And use the docstring from the other PR here.

File: pylint/config/option_manager_mixin.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply pylint/config/option_manager_mixin.py:274]: Yes sorry, I also rebased some commit from the other PR.

File: tests/config/test_config.py [hunk 3]
```diff
@@ -99,9 +99,7 @@ def test_can_read_toml_rich_types(tmp_path: PosixPath) -> None:
     check_configuration_file_reader(config_file)
 
 
-def test_can_read_env_variable(tmp_path: PosixPath) -> None:
-    # Check that we can read the "regular" INI .pylintrc file
-    # if it has an environment variable.
+def test_can_read_toml_env_variable(tmp_path: PosixPath) -> None:
```
Comment (DanielNoord) [inline tests/config/test_config.py:102]: Can't we add a little docstring here?

File: tests/config/test_config.py [hunk 4]
```diff
@@ -99,9 +99,8 @@ def test_can_read_toml_rich_types(tmp_path: PosixPath) -> None:
...
+    """We can read a properly formatted toml file."""
```
Comment (DanielNoord) [inline tests/config/test_config.py:103]:
```suggestion
    """We can read and open a properly formatted toml file."""
```

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**